### PR TITLE
Selective install with nodeSelector

### LINKED ISSUE #10: Ability to selectively install cri-resource-manager (nodes, worker groups)
Allow to specify which nodes or worker groups we want to deploy CRI-RM.

It should be enough to pass nodeSelector labels similar to as gvisor is doing [here](https://github.com/gardener/gardener-extension-runtime-gvisor/blob/master/charts/internal/gvisor-installation/templates/daemonset-containerd.yaml#L26)

daemonset:
```
....
template:
  spec:
      serviceAccountName: cri-resmgr
      automountServiceAccountToken: false
      hostPID: true
      hostIPC: true
      nodeSelector:
{{ toYaml .Values.config.nodeSelector | indent 8 }}
      containers:
...
```

the used like this values.yaml 
```yaml
nodeSelector:
   worker.gardener.cloud/cri-name: containerd
   worker.gardener.cloud/pool: cri-resmgr-pool
```

``worker.gardener.cloud/pool=local`` is set according the name of ``workerGroup`` specified in shoot.

There is also:
``worker.garden.sapcloud.io/group=local`` but I don't know how that relates to above.

### PR DESCRIPTION
**Pull request by ppalucki**
Selective installation using nodeSelector (based on #61) - TO be discussed and tested

resolves #10

### CODE REVIEW

File: charts/internal/cri-resmgr-installation/values.yaml [hunk 1]
```diff
@@ -13,27 +13,30 @@
...
-#         Active: "fallback"
-#   dynamic:
-#     default: |
-#       policy:
-#         Active: "default"
-#     nodeFoo: |
-#       policy:
-#         Active: "nodeFoo"
+### Comment before merging to master!
+### - "images" will be taken from from chart/images.{go,yaml} should be enough and only one source of truth.
```
Comment (mmucek95) [inline charts/internal/cri-resmgr-installation/values.yaml:17]: "images" will be taken from from chart/images.{go,yaml} **and** should be enough and only one source of truth

File: charts/internal/cri-resmgr-installation/values.yaml [hunk 2]
```diff
@@ -13,27 +13,30 @@
...
+###            or can be overwritten with imageVectorOverwrite in ControllerDeployment
+### - "configs" will be provided from configMap or Shoot.Spec.Extensions.providerConfig 
+### - "nodeSelector" provided by spec.Shoot.providerConfig and default one forcing of using containerd as CRI runtime
```
Comment (mmucek95) [inline charts/internal/cri-resmgr-installation/values.yaml:20]: Missing "will be"

File: pkg/controller/lifecycle/actuator.go [hunk 3]
```diff
@@ -41,6 +43,36 @@ import (
...
+	var providerConfig *runtime.RawExtension
+	var criResMgrConfig *CriResMgrConfig
+
+	// Get providerConfig for our extension
+	for _, extension := range extensions {
+		if extension.Type == consts.ExtensionType {
+			providerConfig = extension.ProviderConfig
+		}
+	}
+	// If found, the parse and unmarshal
```
Comment (mmucek95) [inline pkg/controller/lifecycle/actuator.go:65]: s/the/then

**Review by mmucek95 — APPROVED** [review]

File: charts/internal/cri-resmgr-installation/values.yaml (hunk 1 above)
Comment (Damenus) [reply charts/internal/cri-resmgr-installation/values.yaml]: Done

File: charts/internal/cri-resmgr-installation/values.yaml (hunk 2 above)
Comment (Damenus) [reply charts/internal/cri-resmgr-installation/values.yaml]: Added

File: pkg/controller/lifecycle/actuator.go (hunk 3 above)
Comment (Damenus) [reply pkg/controller/lifecycle/actuator.go]: Changed

**Comment by ppalucki** [comment]
@Damenus - I approved that already, but we agreed to also add ability to set those default lables in CtrlDeployment (as default for all the shoot) - is it implemented on you want to add this in another PR?

**Comment by Damenus** [comment]
> @Damenus - I approved that already, but we agreed to also add ability to set those default lables in CtrlDeployment (as default for all the shoot) - is it implemented on you want to add this in another PR?

It is a good idea. We can merge now and this ability I will add this in the next PR.

**Pull request merged**